src/com/facebook/buck/android/ReplaceManifestPlaceholdersStep.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2012-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.android;
+
+import com.facebook.buck.step.ExecutionContext;
+import com.facebook.buck.step.Step;
+import com.facebook.buck.step.StepExecutionResult;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Replaces placeholders in the android manifest.
+ */
+public class ReplaceManifestPlaceholdersStep implements Step {
+
+  private final Path androidManifest;
+  private final Path replacedManifest;
+  private final ImmutableMap<String, String> manifestEntries;
+
+  public ReplaceManifestPlaceholdersStep(
+      Path androidManifest,
+      Path replacedManifest,
+      ImmutableMap<String, String> manifestEntries) {
+    this.androidManifest = androidManifest;
+    this.replacedManifest = replacedManifest;
+    this.manifestEntries = manifestEntries;
+  }
+
+  @Override
+  public StepExecutionResult execute(ExecutionContext context) throws InterruptedException {
+    try {
+      Charset charset = StandardCharsets.UTF_8;
+      String content = new String(Files.readAllBytes(androidManifest), charset);
+      String replaced = replacePlaceholders(content, manifestEntries);
+      Files.write(replacedManifest, replaced.getBytes(charset));
+    } catch (IOException e) {
+      context.logError(e, "Could not replace manifest placeholders.");
+      return StepExecutionResult.ERROR;
+    }
+    return StepExecutionResult.SUCCESS;
+  }
+
+  @Override
+  public String getShortName() {
+    return "replace_manifest_placeholders";
+  }
+
+  @Override
+  public String getDescription(ExecutionContext context) {
+    return getShortName();
```

Comment:
Can you make this look like a shell command with input and output?  like "shortCommand SOURCE_PATH > OUTPUT_PATH"